lib.nix

```diff
@@ -192,9 +189,9 @@ let
         input =
           if inputTakeFrom == "nickel-nix-internals"
           then flakeInputs.${inputTakeFrom}
-          else if inputTakeFrom == "nixpkgs"
-          then flakeInputs.${inputTakeFrom}.legacyPackages.${system}
-          else flakeInputs.${inputTakeFrom}.packages.${system};
+          else
+            flakeInputs.${inputTakeFrom}.legacyPackages.${system}
+            or flakeInputs.${inputTakeFrom}.packages.${system};
```

Comment:
I think this can, in theory, have a different semantics than the original code if the input is different from `nixpkgs` but defines both `legacyPackages` and `packages`. In that case, the old code would select the one from `packages`, will the new code will select the one from `legacyPackages`. However, in practice, I have no idea if this situation even makes sense: can it happen that another flake than `nixpkgs` defines a `legacyPackages` output? Even in this case, I'm not sure the previous behavior was the right one either. I just wanted to point this out

Reply:
Any flake can in theory define both `packages` and `legacyPackages`, but in practice I'm yet to see non-`nixpkgs` flake with `legacyPackages`, because the difference between is quite obscure. That said, hardcoding `nixpkgs` seems worse because if such flake is used somewhere, there would be no way of using it here.